instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java

```diff
@@ -18,6 +18,19 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 // TODO In search for a better home package
+
+/**
+ * Base class for implementing Tracers for HTTP servers. It has 3 types that must be specified by
+ * subclasses:
+ *
+ * @param <REQUEST> - The specific type for HTTP requests
+ * @param <RESPONSE> - The specific type for HTTP responses
+ * @param <CONNECTION> - The specific type of HTTP connection, used to get peer address information
+ *     and HTTP flavor. Use Void if your subclass does not need the connection for peerHostIP,
+ *     peerPort, or flavor.
```

Comment:
can you check if there any subclasses that don't use CONNECTION, I'm not sure if makes sense to use Void for this (unlike STORAGE)

Reply:
> can you check if there any subclasses that don't use CONNECTION, I'm not sure if makes sense to use Void for this (unlike STORAGE)

Good call. They all do.  I'll remove the `Void` bit.